# **EDA**


In [18]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import json

df = pd.read_csv("C:/Users/valen/OneDrive/Documentos/workshop_1/dataset/candidates.csv", sep=";")

In [3]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   First Name                 50000 non-null  object
 1   Last Name                  50000 non-null  object
 2   Email                      50000 non-null  object
 3   Application Date           50000 non-null  object
 4   Country                    50000 non-null  object
 5   YOE                        50000 non-null  int64 
 6   Seniority                  50000 non-null  object
 7   Technology                 50000 non-null  object
 8   Code Challenge Score       50000 non-null  int64 
 9   Technical Interview Score  50000 non-null  int64 
dtypes: int64(3), object(7)
memory usage: 3.8+ MB


First Name   Last Name                      Email Application Date  \
0  Bernadette   Langworth        leonard91@yahoo.com       2021-02-26   
1      Camryn    Reynolds        zelda56@hotmail.com       2021-09-09   
2       Larue      Spinka   okey_schultz41@gmail.com       2020-04-14   
3        Arch      Spinka     elvera_kulas@yahoo.com       2020-10-01   
4       Larue  Altenwerth  minnie.gislason@gmail.com       2020-05-20   

   Country  YOE  Seniority                         Technology  \
0   Norway    2     Intern                      Data Engineer   
1   Panama   10     Intern                      Data Engineer   
2  Belarus    4  Mid-Level                     Client Success   
3  Eritrea   25    Trainee                          QA Manual   
4  Myanmar   13  Mid-Level  Social Media Community Management   

   Code Challenge Score  Technical Interview Score  
0                     3                          3  
1                     2                         10  
2                    10                          9  
3                     7                          1  
4                     9                          7

El csv tiene 10 columnas con diferentes tipos de datos. La mayoría son texto o varchar, excepto YOE, Code Challenge Score y Technical Interview Score que son enteros.

In [4]:
print(df.duplicated().sum())

0


After reviewing the dataset, we can say there are not duplicated records on it.

In [5]:
df.isnull().sum()

First Name                   0
Last Name                    0
Email                        0
Application Date             0
Country                      0
YOE                          0
Seniority                    0
Technology                   0
Code Challenge Score         0
Technical Interview Score    0
dtype: int64

Not null values on the dataset.

In [6]:
df.describe()

YOE  Code Challenge Score  Technical Interview Score
count  50000.000000          50000.000000               50000.000000
mean      15.286980              4.996400                   5.003880
std        8.830652              3.166896                   3.165082
min        0.000000              0.000000                   0.000000
25%        8.000000              2.000000                   2.000000
50%       15.000000              5.000000                   5.000000
75%       23.000000              8.000000                   8.000000
max       30.000000             10.000000                  10.000000

This table shows a statistical summary of a DataFrame for three columns: 'Years of Experience' (YOE), 'Code Challenge Score', and 'Technical Interview Score'. Here are some insights:

- Count: Each column contains data for 50,000 entries, suggesting no missing values in these fields.
- Mean: On average, candidates have around 15.29 years of experience. The average scores for both the Code Challenge and Technical Interview are close to 5.
- Std (Standard Deviation): The years of experience have a standard deviation of approximately 8.83, indicating a wide range of experience levels among candidates. The Code Challenge and Technical Interview scores have a standard deviation of around 3.17, showing moderate variability in scores.
- Min: The minimum value for all three columns is 0, indicating that there are entries with no experience or no score given for some candidates.
- 25% (1st Quartile): 25% of candidates have 8 or fewer years of experience and scored 2 or lower in both the Code Challenge and Technical Interview.
- 50% (Median): The median years of experience is 15, and the median score for both the Code Challenge and Technical Interview is 5, which is also the maximum possible score.
- 75% (3rd Quartile): 75% of candidates have up to 23 years of experience and scored 8 or lower in the assessments.
- Max: The maximum years of experience recorded is 30, and the highest score for both the Code Challenge and Technical Interview is 10.


In [15]:
unique_technologies = df['Technology'].unique()
unique_technologies

array(['Data Engineer', 'Client Success', 'QA Manual',
       'Social Media Community Management', 'Adobe Experience Manager',
       'Sales', 'Mulesoft', 'DevOps', 'Development - CMS Backend',
       'Salesforce', 'System Administration', 'Security',
       'Game Development', 'Development - CMS Frontend',
       'Security Compliance', 'Development - Backend', 'Design',
       'Business Analytics / Project Management',
       'Development - Frontend', 'Development - FullStack',
       'Business Intelligence', 'Database Administration',
       'QA Automation', 'Technical Writing'], dtype=object)

Based on the output, it seems there are 24 unique technology specializations or roles, including 'Data Engineer', 'Client Success', 'QA Manual', and others, suggesting a diverse set of technology-related fields among the candidates in the dataset